projects/ui/src/lib/components/po-stepper/po-stepper.component.ts

```diff
@@ -135,15 +135,17 @@ export class PoStepperComponent extends PoStepperBaseComponent implements AfterC
 
   changeStep(stepIndex: number, step?: PoStepComponent): void {
     const isDifferentStep = (!this.currentActiveStep || step.id !== this.currentActiveStep.id);
-
-    if (this.usePoSteps && isDifferentStep && this.allowNextStep(stepIndex)) {
-      this.controlStepsStatus(stepIndex, step);
-
-      this.onChangeStep.emit(step);
-    } else if (!this.usePoSteps && stepIndex !== this.currentStepIndex && this.allowNextStep(stepIndex)) {
-      // if para tratamento do modelo antigo do po-stepper
-      this.onChangeStep.emit(stepIndex + 1);
-    }
+    this.allowNextStep(stepIndex).subscribe(nextStepAllowed => {
```

Comment:
É importante fazer o `unsubscribe` do `allowNextStep`. Neste link dá algumas ideias de como isso pode ser feito:
https://medium.com/angular-in-depth/the-best-way-to-unsubscribe-rxjs-observable-in-the-angular-applications-d8f9aa42f6a0